src/libraries/System.Private.CoreLib/src/System/IO/Strategies/WindowsFileStreamStrategy.cs

```diff
@@ -6,7 +6,7 @@
 using Microsoft.Win32.SafeHandles;
 using System.Runtime.CompilerServices;
 
-namespace System.IO
+namespace System.IO.Strategies
```

Comment:
FWIW, I would prefer if all the "strategies" were just nested types within FileStream.  They're not usable nor meant to be used from outside of it, they needn't be "internal" and visible to the rest of the assembly, etc.

Reply:
IMO they are too big to be nested. The formatting would look just bad (it's a personal opinion)

Reply:
I guess it would look like this:
```cs
namespace System.IO
{
    public partial class FileStream : Stream
    {
        private sealed class BufferedFileStreamStrategy : FileStreamStrategy
        {
            ...
        }
    }
}
```

I'm on the fence on this subject, I think that is nice that we could prevent visibility of the strategies but if it is too cumbersome for the code I don't think that is worth it IMO.

Reply:
> IMO they are too big to be nested. The formatting would look just bad

I disagree.  The only thing it adds is one extra level of indentation.  And in exchange, you get:
a) Shorter type names everywhere they're mentioned because you no longer need to include "FileStream" in all the type names, e.g. it's just "Strategy" and "BufferedStrategy" rather than "FileStreamStrategy" and "BufferedFileStreamStrategy".  That simplification alone makes it more appealing to me.
b) Everywhere else in corelib, you don't see System.IO.Strategies in IntelliSense / completion lists, and no one else is able to even try to create one of these things because they're invisible.